src/main/java/backend/stub/DummyRepoState.java

```diff
@@ -400,6 +398,38 @@ protected TurboUser deleteUser(String idString) {
         return newLabels.stream().map(new Label()::setName).collect(Collectors.toList());
     }
 
+    protected final Issue setMilestone(int issueId, Optional<Integer> milestone) {
+        ImmutablePair<TurboIssue, IssueMetadata> mutables = produceMutables(issueId);
+        TurboIssue toSet = mutables.getLeft();
+        IssueMetadata metadataOfIssue = mutables.getRight();
+        List<TurboIssueEvent> eventsOfIssue = metadataOfIssue.getEvents();
+
+        if (toSet.getMilestone().isPresent()) {
+            int milestoneOfIssue = toSet.getMilestone().get();
+            eventsOfIssue.add(new TurboIssueEvent(new User().setLogin("test"),
+                    IssueEventType.Demilestoned,
+                    new Date()).setMilestoneTitle(milestones.get(milestoneOfIssue).getTitle()));
+        }
+
+        if (milestone.isPresent()) {
+            eventsOfIssue.add(new TurboIssueEvent(new User().setLogin("test"),
+                    IssueEventType.Milestoned,
+                    new Date()).setMilestoneTitle(milestones.get(milestone.get()).getTitle()));
+            toSet.setMilestoneById(milestone.get());
+        } else {
+            toSet.removeMilestone();
+        }
+        toSet.setUpdatedAt(LocalDateTime.now());
+
+        markUpdatedEvents(toSet, IssueMetadata.intermediate(eventsOfIssue, metadataOfIssue.getComments(), "", ""));
+
+        Issue newIssue = new Issue();
+        if (milestone.isPresent()) {
+            newIssue.setMilestone(new Milestone().setNumber(milestone.get()));
+        }
+        return newIssue;
```

Comment:
The high level steps are not immediately clear. May be add `//comments` to each block of code?